WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -0,0 +1,204 @@
+import Foundation
+import AutomatticTracks
+
+class RegisterDomainCoordinator {
+
+    // MARK: Type Aliases
+
+    typealias DomainPurchasedCallback = ((UIViewController, String) -> Void)
+    typealias DomainAddedToCartCallback = ((UIViewController, String) -> Void)
+
+    // MARK: Variables
+
+    private let crashLogger: CrashLogging
+
+    var site: Blog?
+    var domainPurchasedCallback: DomainPurchasedCallback?
+    var domainAddedToCartCallback: DomainAddedToCartCallback?
+    var domain: FullyQuotedDomainSuggestion?
+
+    private var webViewURLChangeObservation: NSKeyValueObservation?
+
+    init(site: Blog?,
+         domainPurchasedCallback: RegisterDomainCoordinator.DomainPurchasedCallback? = nil,
+         crashLogger: CrashLogging = .main) {
+        self.site = site
+        self.domainPurchasedCallback = domainPurchasedCallback
+        self.crashLogger = crashLogger
+    }
+
+    // MARK: Public Functions
+
+    func createCart(onSuccess: @escaping () -> (),
+                    onFailure: @escaping () -> ()) {
+        guard let domain else { return }
+        let siteID = site?.dotComID?.intValue
+        let proxy = RegisterDomainDetailsServiceProxy()
+        proxy.createPersistentDomainShoppingCart(siteID: siteID,
+                                                 domainSuggestion: domain.remoteSuggestion(),
+                                                 privacyProtectionEnabled: domain.supportsPrivacy ?? false,
+                                                 success: { _ in
+            onSuccess()
+        },
+                                                 failure: { _ in
+            onFailure()
+        })
+    }
+
+    func presentWebViewForNoSite(on viewController: UIViewController) {
+        guard let domain,
+              let url = URL(string: Constants.noSiteCheckoutWebAddress) else {
+            crashLogger.logMessage("Failed to present domain checkout webview for no-site.",
+                                   level: .error)
+            return
+        }
+
+        presentCheckoutWebview(on: viewController,
+                               domainSuggestion: domain,
+                               url: url,
+                               title: TextContent.checkoutTitle,
+                               shouldPush: true)
+    }
+
+    func presentWebViewForCurrentSite(on viewController: UIViewController) {
```

Comment:
❓ I'm curious if we should have a single `presentCheckoutWebView` method that handles site vs no-site logic based on whether `self.site` is `nil` or not. Because right now, it is **programmatically** possible to:
- Call `presentWebViewForCurrentSite` even though `self.site` is `nil`. Which would result in an error.
- Call `presentWebViewForNoSite` even though `self.site` is not `nil`. This wouldn't result in an error, but it might be confusing. We are presenting the Checkout web view for no-site even though the coordinator has a site?